snapshotter/utils/rpc.py

```diff
@@ -417,16 +416,26 @@ async def f(node_idx):
                 )
 
                 if from_address:
-                    payload['from'] = from_address
-
-                data = await node['web3_client_async'].eth.call(payload)
+                    payload["from"] = from_address
 
-                decoded_data = node['web3_client_async'].codec.decode_abi(
-                    output_type, HexBytes(data),
+                data = await node["web3_client_async"].eth.call(
+                    payload, block_identifier=block, state_override=overrides
+                    )
+                
+                # if we're doing a state override call, at time of writing it means grabbing tick data
+                # more efficient to use eth_abi to decode rather than web3 codec
```

Comment:
Better to write this along with some examples in function docstrings at the top.